# HG changeset patch
# User Max Brister <max@2bass.com>
# Date 1335384732 21600
#      Wed Apr 25 14:12:12 2012 -0600
# Node ID 3a10d5bdada85536de491694f07c7f7c1b90f261
# Parent  e97ec01d4157760053ca41a701d90fe42cbf0dbb
src/symtab.yy (ABORT_PARSE): Prevent multiple parse errors in eval (bug #35154)

diff --git a/src/oct-parse.yy b/src/oct-parse.yy
--- a/src/oct-parse.yy
+++ b/src/oct-parse.yy
@@ -368,17 +368,18 @@ make_statement (T *arg)
 
 #define ABORT_PARSE \
   do \
     { \
       global_command = 0; \
       yyerrok; \
       if (! parser_symtab_context.empty ()) \
         parser_symtab_context.pop (); \
-      if (interactive || forced_interactive) \
+      if ((interactive || forced_interactive)   \
+          && ! get_input_from_eval_string)      \
         YYACCEPT; \
       else \
         YYABORT; \
     } \
   while (0)
 
 %}
 
